calibration_utils.py

```diff
@@ -110,16 +110,30 @@ def calibrate(self, board_config, filepath, square_size, mrk_size, squaresX, squ
             self.aruco_dictionary)
 
         # parameters = aruco.DetectorParameters_create()
-        assert mrk_size != None,  "ERROR: marker size not set"
         combinedCoverageImage = None
+        resizeWidth, resizeHeight = 0, 0
+        assert mrk_size != None,  "ERROR: marker size not set"
+        for camera in board_config['cameras'].keys():
+            cam_info = board_config['cameras'][camera]
+            images_path = filepath + '/' + cam_info['name']
+            image_files = glob.glob(images_path + "/*")
+            image_files.sort()
+            for im in image_files:
+                frame = cv2.imread(im)
+                height, width, _ = frame.shape
+                widthRatio = resizeWidth / width
+                heightRatio = resizeHeight / height
+                if (widthRatio > 0.8 and heightRatio > 0.8 and widthRatio <= 1.0 and heightRatio <= 1.0) or (widthRatio > 1.2 and heightRatio > 1.2) or (resizeHeight == 0):
+                    resizeWidth = width
+                    resizeHeight = height
+                break
         for camera in board_config['cameras'].keys():
             cam_info = board_config['cameras'][camera]
             print(
                 '<------------Calibrating {} ------------>'.format(cam_info['name']))
             images_path = filepath + '/' + cam_info['name']
             ret, intrinsics, dist_coeff, _, _, size, coverageImage = self.calibrate_intrinsics(
                 images_path, cam_info['hfov'])
-            # coverageImages[cam_info['name']] = coverageImage
             cam_info['intrinsics'] = intrinsics
             cam_info['dist_coeff'] = dist_coeff
             cam_info['size'] = size # (Width, height)
```

Comment:
you can get the size here. and use it to calculate the resizedWidth and height. No need for the newly introduced block above.